lib/python/qmk/cli/cformat.py

```diff
@@ -65,10 +65,20 @@ def cformat_run(files):
         return False
 
 
-def filter_files(files):
+def filter_files(files, core_only=False):
     """Yield only files to be formatted and skip the rest
     """
-    for file in files:
+    if core_only:
+        # Filter non-core files
+        for index, file in enumerate(files):
+            # The following statement checks each file to see if the file path is
+            # - in the core directories
+            # - not in the ignored directories
+            if not any(i in str(file) for i in core_dirs) or any(i in str(file) for i in ignored):
+                files[index] = None
+                cli.log.debug("Skipping non-core file %s, as '--core-only' is used.", file)
+
+    for file in [file for file in files if file is not None]:
         if file.name.split('.')[-1] in c_file_suffixes:
```

Comment:
This avoids iterating over the files twice.

```suggestion
    for file in files:
        if file and file.name.split('.')[-1] in c_file_suffixes:
```